test/models/merged_ticket_test.rb

```diff
@@ -0,0 +1,62 @@
+require 'test_helper'
+
+class TicketMergeTest < ActiveSupport::TestCase
+  
+  test 'merging two tickets' do
+    client_user = User.where(agent: false).first
+    agent_user = User.where(agent: true).first
+    first_ticket = Ticket.create subject: "First ticket", content: "I've got a question ...", user_id: client_user.id
+    Timecop.travel 2.minutes.from_now
+    first_reply = first_ticket.replies.create content: "Please send me some more info about ...", user_id: agent_user.id
+    Timecop.travel 20.minutes.from_now
+    second_ticket = Ticket.create subject: "Second ticket", content: "I forgot to mention ...", user_id: client_user.id
+    Timecop.travel 1.hour.from_now
+    
+    second_ticket_updated_at_before_merge = second_ticket.updated_at
+    
+    separate_tickets = [first_ticket, second_ticket]
+    merged_ticket = MergedTicket.from separate_tickets
+    
+    assert_equal 2, merged_ticket.replies.count
+    assert_equal merged_ticket.id, first_ticket.id
+    assert merged_ticket.replies.pluck(:content).include? "Please send me some more info about ..."
+    assert merged_ticket.replies.pluck(:content).include? "I forgot to mention ..."
+    assert merged_ticket.replies.order(:created_at).first.user == agent_user
+    assert merged_ticket.replies.order(:created_at).last.user == client_user
+    assert_equal merged_ticket.replies.order(:created_at).last.created_at.to_i, second_ticket.created_at.to_i
+    assert_equal merged_ticket.replies.order(:created_at).last.updated_at.to_i, second_ticket_updated_at_before_merge.to_i
+  end
+  
+  test 'merging two tickets and providing an agent who is performing the merge' do
+    client_user = User.where(agent: false).first
+    agent_user = User.where(agent: true).first
+    first_ticket = Ticket.create subject: "First ticket", content: "I've got a question ...", user_id: client_user.id
+    Timecop.travel 2.minutes.from_now
+    second_ticket = Ticket.create subject: "Second ticket", content: "I forgot to mention ...", user_id: client_user.id
+    
+    merged_ticket = MergedTicket.from [first_ticket, second_ticket], current_user: agent_user
+    second_ticket.reload
+    
+    assert_equal 1, merged_ticket.replies.count
+    assert second_ticket.replies.last.user == agent_user
+    assert second_ticket.replies.last.internal == true
+    assert second_ticket.replies.last.content.include? first_ticket.id.to_s
+    assert second_ticket.status == 'closed'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.